### PR TITLE
[PCM-1912] reject non acd calls

### PR DESCRIPTION
**Pull request by jensengar**

### CODE REVIEW

**Review by maxwellmooney13 — APPROVED** [review]

**Review by zservies — APPROVED** [review]

File: test/unit/sessions/softphone-session-handler.test.ts [hunk 1]
```diff
@@ -233,7 +233,7 @@ describe('acceptSession()', () => {
     const startMediaSpy = jest.spyOn(mockSdk.media, 'startMedia');
 
     const element = {};
-    const volume = mockSdk._config.defaults.audioVolume = 67;
+    const volume = mockSdk._config.defaults!.audioVolume = 67;
```
Comment (davidmorton-gen) [inline test/unit/sessions/softphone-session-handler.test.ts:236]: Codacy has a fix for the issue: [Forbidden non-null assertion.](https://app.codacy.com/gh/MyPureCloud/genesys-cloud-webrtc-sdk/pullRequest?prid=9630720)
```suggestion
    const volume = mockSdk._config.defaults?.audioVolume = 67;
```

File: test/unit/sessions/softphone-session-handler.test.ts [hunk 2]
```diff
@@ -267,9 +267,9 @@ describe('acceptSession()', () => {
     const defaultElement = {};
     const defaultStream = new MockStream({ audio: true });
 
-    mockSdk._config.defaults.audioElement = defaultElement as any;
-    mockSdk._config.defaults.audioStream = defaultStream as any;
-    const volume = mockSdk._config.defaults.audioVolume = 67;
+    mockSdk._config.defaults!.audioElement = defaultElement as any;
```
Comment (davidmorton-gen) [inline test/unit/sessions/softphone-session-handler.test.ts:270]: Codacy has a fix for the issue: [Forbidden non-null assertion.](https://app.codacy.com/gh/MyPureCloud/genesys-cloud-webrtc-sdk/pullRequest?prid=9630720)
```suggestion
    mockSdk._config.defaults?.audioElement = defaultElement as any;
```

File: src/sessions/softphone-session-handler.ts [hunk 3]
```diff
@@ -539,11 +531,22 @@ export default class SoftphoneSessionHandler extends BaseSessionHandler {
...
+    if (participant.purpose === 'user') {
+      return this._rejectUcCall(pendingSession.conversationId, participant.id);
+    }
+
     return this.patchPhoneCall(pendingSession.conversationId, participant.id, {
       state: CommunicationStates.disconnected
     });
   }
 
+  _rejectUcCall (conversationId: string, participantId: string): Promise<any> {
```
Comment (davidmorton-gen) [inline src/sessions/softphone-session-handler.ts:543]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/MyPureCloud/genesys-cloud-webrtc-sdk/pullRequest?prid=9630720)
```suggestion
  _rejectUcCall (conversationId: string, participantId: string): Promise<unknown> {
```

File: src/sessions/softphone-session-handler.ts [hunk 4]
```diff
@@ -523,14 +523,6 @@ export default class SoftphoneSessionHandler extends BaseSessionHandler {
   }
 
   async rejectPendingSession (pendingSession: IPendingSession): Promise<any> {
-    if (!this.hasActiveSession()) {
-      return super.rejectPendingSession(pendingSession);
-    }
```
Comment (djhouseknecht) [inline src/sessions/softphone-session-handler.ts:528]: Can we send a PATCH even if we don't have a session?

Comment (jensengar) [reply src/sessions/softphone-session-handler.ts:528]: yes.

**Review by djhouseknecht — APPROVED** [review]

**Pull request merged**